### PR TITLE
Replace remove_conversation_turns with summarize_turns standard tool

### PR DESCRIPTION
**Pull request by snimu**
## Description

Overhaul context dropping in RLMEnv:
- Rename r`emove_conversation_turns` → `summarize_turns` and move from root-REPL tool to standard tool (called via normal tool calling)
- Summaries are cumulative: each call appends [Turns X-Y] section
- Summary injected as <SUMMARY> block into first remaining assistant message instead of being written to .summaries file
- .messages file remains unaffected (full uncompacted history)
- Replace `compaction_*` metrics with `summarize_*` metrics including char/token compression tracking
- Track per-turn prompt_tokens for accurate token-drop measurement
- 38 new tests replacing 16 old ones

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Test improvement

## Testing

- [x] All existing tests pass when running `uv run pytest` locally.
- [x] New tests have been added to cover the changes

## Checklist
- [x] My code follows the style guidelines of this project as outlined in [AGENTS.md](https://github.com/PrimeIntellect-ai/verifiers/blob/main/AGENTS.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

---

> [!NOTE]
> **Medium Risk**
> Changes how conversation context is truncated and re-injected into prompts, which can affect model behavior and any downstream consumers of stored messages/metrics. Also renames/removes previous tool and metrics, so integrations relying on `remove_conversation_turns` or `compaction_*` will break if not updated.
> 
> **Overview**
> Replaces the root-only `remove_conversation_turns` mechanism with an opt-in standard tool, `summarize_turns`, controlled by `enable_summarization`, which drops older turns while maintaining a **cumulative** summary in state.
> 
> Summaries are now injected into the first remaining assistant message as a `<SUMMARY>` block (handling both string and list message content) and `.messages` serialization is updated to strip these injected blocks so message history remains clean.
> 
> Removes `.summaries` uploading and `compaction_*` state/metrics, replacing them with `summarize_*` metrics including dropped character tracking and a compression ratio; tests are overhauled to cover tool registration, prompt notes, injection behavior, edge cases, and the new metrics.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 042d6a4744d208c72b66c128c6f428b1ad6a32e1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: verifiers/envs/experimental/rlm_env.py [hunk 1]
```diff
@@ -3993,6 +3863,174 @@ async def call_python_repl(self, code: str, state: Any) -> str:
...
+                total_chars += len(content)
+            elif isinstance(content, list):
+                for part in content:
+                    if isinstance(part, dict):
+                        total_chars += len(str(part.get("text", "")))
+                    else:
+                        total_chars += len(str(part))
+            else:
+                total_chars += len(str(content))
+        return total_chars
```
[the remark this answers is not in the document]
Comment (snimu) [reply verifiers/envs/experimental/rlm_env.py:4040]: this is intended; the summary is part of what the model sees and should thus be counted

File: verifiers/envs/experimental/rlm_env.py [hunk 2]
```diff
@@ -2258,6 +2268,11 @@ class RLMEnv(vf.StatefulToolEnv):
         enable_sub_llms: If True (default), the root model can call sub-LLMs via
                    ``llm_batch()``. If False, ``llm_batch`` is not registered as a
                    tool and all sub-LLM-related prompt sections are omitted.
+        enable_summarization: If True, the root model gets a ``summarize_turns``
+                   tool to drop old conversation turns and replace them with a
+                   cumulative summary. Defaults to False.
+        min_turns_in_context: Minimum turns that must remain after summarization
+                   (default: 3). Only has effect when enable_summarization=True.
```
[the remark this answers is not in the document]
Comment (snimu) [reply verifiers/envs/experimental/rlm_env.py:2275]: the doc files are oneliners and that's fine. the changes are documented in the docstring so that's fine.

File: verifiers/envs/experimental/rlm_env.py [hunk 3]
```diff
@@ -3993,6 +3874,171 @@ async def call_python_repl(self, code: str, state: Any) -> str:
...
+                total_chars += len(content)
+            elif isinstance(content, list):
+                for part in content:
+                    if isinstance(part, dict):
+                        total_chars += len(str(part.get("text", "")))
+                    else:
+                        total_chars += len(str(part))
+            else:
+                total_chars += len(str(content))
+        return total_chars
```
[the remark this answers is not in the document]
Comment (snimu) [reply verifiers/envs/experimental/rlm_env.py:4040]: dumb edge-case.
